submissions/bmukha/frogger/app.js

```diff
@@ -0,0 +1,149 @@
+let highScore = localStorage.getItem('highScore') ?? 0;
+let currentScore = 0;
+const highHolder = document.querySelector('.high');
+const currentHolder = document.querySelector('.current');
+const infoHolder = document.querySelector('.info');
+
+const getRandomNumber = (min, max) =>
+  Math.floor(Math.random() * (max - min) + min);
+
+const directions = ['left', 'right'];
+
+const getRandomDirection = () =>
+  directions[Math.floor(Math.random() * directions.length)];
+
+canvas = {
+  width: 505,
+  height: 606,
+};
+
+class Enemy {
+  constructor(startY, mult) {
+    this.width = 101;
+    this.height = 171;
+    this.direction = getRandomDirection();
+    this.sprite = `images/enemy-bug-${this.direction}.png`;
+    this.minX = -this.width;
+    this.maxX = canvas.width + this.width;
+    this.startX = this.direction === 'right' ? this.minX : this.maxX;
+    this.startY = startY;
+    this.currX = this.startX;
+    this.currY = this.startY;
+    this.mult = this.direction === 'right' ? mult : -mult;
+  }
+  update(dt) {
+    if (
+      player.currX >= this.currX &&
+      player.currX <= this.currX + this.width &&
+      player.currY >= this.currY &&
+      player.currY <= this.currY + 85
```

Comment:
(1) classes should not not refer to any global variables, like global variable `player`, which is an instance of `Player` class

(referring to global constants and globals provided by the gaming platform like `Resources` is OK);

Hint: pass `Player` instance as an argument to every enemy

(2) Separation of Concerns principle should be followed

(e.g. `update` method does only rendering and doesn't contain any unrelated **inline** code; for example collision check is defined as a dedicated method and only called from inside `update`)